x/oracle/abci.go

```diff
@@ -108,11 +106,15 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper, experimental bool) error {
 		k.SlashAndResetMissCounters(ctx)
 	}
 
-	// Prune historic prices every prune period
-	if isPeriodLastBlock(ctx, params.PrunePeriod) && experimental {
-		pruneBlock := uint64(ctx.BlockHeight()) - params.PrunePeriod
-		for _, v := range params.HistoricAcceptList {
-			k.DeleteHistoricPrice(ctx, v.String(), pruneBlock)
+	// Prune historic prices and medians outside pruning period determined by
+	// the stamp period multiplied by the max stamps.
+	if experimental {
+		pruneHistoricBlock := uint64(ctx.BlockHeight()) - (params.HistoricStampPeriod * params.MaximumPriceStamps)
+		pruneMedianBlock := uint64(ctx.BlockHeight()) - (params.MedianStampPeriod * params.MaximumMedianStamps)
```

Comment:
how about avoiding prunning every block, and instead batch it, eg once every 100 blocks (10min).

Reply:
I changed it to prune every historic stamp period. Any less frequent would result in us storing more than the maximum price stamps.